src/main/java/com/aws/greengrass/lifecyclemanager/Kernel.java

```diff
@@ -166,6 +166,31 @@ public Kernel() {
         context.put(SERVICE_TYPE_TO_CLASS_MAP_KEY, typeToClassMap);
     }
 
+    /**
+     * Construct the Kernel to be used for connectivity validation.
+     *
+     * @param currentConfig         current device configs
+     * @param deploymentConfig      Map of the configs being merged in from deployment
+     * @param deploymentTimestamp   timestamp of the deployment being merged in
+     */
+    public Kernel(DeviceConfiguration currentConfig, Map<String, Object> deploymentConfig,
+                  long deploymentTimestamp) {
+        context = new Context();
+        config = new Configuration(context);
+        config.copyFrom(currentConfig.getConfig());
+        config.mergeMap(deploymentTimestamp, deploymentConfig);
+        try {
+            config.waitConfigUpdateComplete();
+        } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

**Remember**: Do this only for InterruptedException. Doing this for other types of exceptions may give rise to subtle bugs.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)